Are tests failing on Windows?

Running the tests on windows fail:

```
>>> pytest pandas\tests\tseries\offsets\test_offsets.py -x
――――――――――――――――――――― TestCommon.test_apply[BusinessDay0] ―――――――――――――――――――――

self = <pandas.tests.tseries.offsets.test_offsets.TestCommon object at 0x066D4530>
offset_types = <class 'pandas.tseries.offsets.BusinessDay'>

    def test_apply(self, offset_types):
        sdt = datetime(2011, 1, 1, 9, 0)
        ndt = np_datetime64_compat('2011-01-01 09:00Z')

        for dt in [sdt, ndt]:
            expected = self.expecteds[offset_types.__name__]
            self._check_offsetfunc_works(offset_types, 'apply', dt, expected)

            expected = Timestamp(expected.date())
            self._check_offsetfunc_works(offset_types, 'apply', dt, expected,
>                                        normalize=True)

pandas\tests\tseries\offsets\test_offsets.py:357:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas\tests\tseries\offsets\test_offsets.py:320: in _check_offsetfunc_works
    result = func(dt_tz)
pandas\tseries\offsets.py:85: in wrapper
    result = result.normalize()
pandas\_libs\tslibs\timestamps.pyx:1086: in pandas._libs.tslibs.timestamps.Timestamp.normalize
    normalized_value = normalize_i8_timestamps(
pandas\_libs\tslibs\conversion.pyx:1095: in pandas._libs.tslibs.conversion.normalize_i8_timestamps
    result = _normalize_local(stamps, tz)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   pos = trans.searchsorted(stamps, side='right') - 1
E   ValueError: Buffer dtype mismatch, expected 'int64_t' but got 'int'

pandas\_libs\tslibs\conversion.pyx:1151: ValueError
```

Several tests fail. 

By going back in commit history until the tests pass, it looks like this is caused by #21826.

Still  a problem:

```python
――――――――――――――――――――――― test_fy5253qtr_onoffset_nearest ―――――――――――――――――――――――

    def test_fy5253qtr_onoffset_nearest():
        # GH#19036
        ts = Timestamp('1985-09-02 23:57:46.232550356-0300',
                       tz='Atlantic/Bermuda')
        offset = FY5253Quarter(n=3, qtr_with_extra_week=1, startingMonth=2,
                               variation="nearest", weekday=0)
>       fast = offset.onOffset(ts)

pandas\tests\tseries\offsets\test_fiscal.py:645:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas\tseries\offsets.py:2018: in onOffset
    qtr_lens = self.get_weeks(dt)
pandas\tseries\offsets.py:1992: in get_weeks
    year_has_extra_week = self.year_has_extra_week(dt)
pandas\tseries\offsets.py:2002: in year_has_extra_week
    norm = Timestamp(dt).normalize().tz_localize(None)
pandas\_libs\tslibs\timestamps.pyx:1095: in pandas._libs.tslibs.timestamps.Timestamp.normalize
    normalized_value = normalize_i8_timestamps(
pandas\_libs\tslibs\conversion.pyx:1094: in pandas._libs.tslibs.conversion.normalize_i8_timestamps
    result = _normalize_local(stamps, tz)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   pos = trans.searchsorted(stamps, side='right') - <int64_t>1
E   ValueError: Buffer dtype mismatch, expected 'int64_t' but got 'int'

pandas\_libs\tslibs\conversion.pyx:1150: ValueError
```

--- a/pandas/_libs/tslibs/conversion.pyx
+++ b/pandas/_libs/tslibs/conversion.pyx
@@ -527,7 +527,7 @@ cdef inline void localize_tso(_TSObject obj, tzinfo tz):
     cdef:
         ndarray[int64_t] trans, deltas
         int64_t delta, local_val
-        Py_ssize_t posn
+        Py_ssize_t pos
 
     assert obj.tzinfo is None
 
@@ -782,7 +782,6 @@ def tz_convert(ndarray[int64_t] vals, object tz1, object tz2):
     cdef:
         ndarray[int64_t] utc_dates, tt, result, trans, deltas
         Py_ssize_t i, j, pos, n = len(vals)
-        ndarray[Py_ssize_t] posn
         int64_t v, offset, delta
         npy_datetimestruct dts
 
@@ -1124,7 +1123,8 @@ cdef ndarray[int64_t] _normalize_local(ndarray[int64_t] stamps, object tz):
     cdef:
         Py_ssize_t n = len(stamps)
         ndarray[int64_t] result = np.empty(n, dtype=np.int64)
-        ndarray[int64_t] trans, deltas, pos
+        ndarray[int64_t] trans, deltas
+        Py_ssize_t[:] pos
         npy_datetimestruct dts
 
     if is_utc(tz):

--- a/pandas/_libs/tslibs/period.pyx
+++ b/pandas/_libs/tslibs/period.pyx
@@ -934,7 +934,8 @@ cdef ndarray[int64_t] localize_dt64arr_to_period(ndarray[int64_t] stamps,
     cdef:
         Py_ssize_t n = len(stamps)
         ndarray[int64_t] result = np.empty(n, dtype=np.int64)
-        ndarray[int64_t] trans, deltas, pos
+        ndarray[int64_t] trans, deltas
+        Py_ssize_t[:] pos
         npy_datetimestruct dts
         int64_t local_val
 

--- a/pandas/_libs/tslibs/resolution.pyx
+++ b/pandas/_libs/tslibs/resolution.pyx
@@ -74,7 +74,8 @@ cdef _reso_local(ndarray[int64_t] stamps, object tz):
     cdef:
         Py_ssize_t n = len(stamps)
         int reso = RESO_DAY, curr_reso
-        ndarray[int64_t] trans, deltas, pos
+        ndarray[int64_t] trans, deltas
+        Py_ssize_t[:] pos
         npy_datetimestruct dts
         int64_t local_val
 

